api/apps/api/src/migrations/api/1687796245859-AddProjectFileCacheTable.ts

```diff
@@ -0,0 +1,42 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class AddProjectFileCacheTable1687796245859
+  implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `DROP INDEX "cost_surface_file_cache_scenarioId_artifactType_idx";`,
+    );
+    await queryRunner.query(`DROP TABLE "cost_surface_file_cache"`);
+    await queryRunner.query(
+      `DROP TYPE "cost_surface_file_cache_artifactType_enum"`,
+    );
+
+    await queryRunner.query(`
+      CREATE TYPE "project_template_file_cache_artifactType_enum" AS ENUM('ProjectTemplate');
+    `);
+    await queryRunner.query(`
+      CREATE TABLE "project_template_file_cache" (
+          "id" uuid PRIMARY KEY NOT NULL DEFAULT gen_random_uuid(),
+          "projectId" uuid NOT NULL REFERENCES projects(id) ON DELETE CASCADE ON UPDATE CASCADE,
+          "artifact" oid,
+          "artifactType" "project_template_file_cache_artifactType_enum" NOT NULL,
+          "contentType" character varying NOT NULL,
+          "createdAt" TIMESTAMP NOT NULL DEFAULT now(),
+          "lastModifiedAt" TIMESTAMP NOT NULL DEFAULT now()
```

Comment:
also for the "while refactoring, but trying to keep this to low-hanging fruits" department: maybe this table should have been a generic "artifact_cache" since the beginning - at the project level, but potentially able to deal with scenario-specific artifacts in the future if needed.

If you think this could be a relatively quick change, I'd recommend to rework names throughout (starting from this table, through TS classes and so on) to remove references to "project" and "template" and treat this as a generic cache of artifacts, that happens to hold project-level templates for cost surface shapefiles and for grid shapefiles.

Or leave as is and we can ship it, as it'll do what it needs to do (I haven't actually tested the functionality, but if it isn't fully working yet, it should not be far from it).